debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnectorTask.java

```diff
@@ -86,7 +86,7 @@ public String version() {
                 context.offsetStorageReader(),
                 new SqlServerOffsetContext.Loader(connectorConfig));
         TaskOffsetContext<SqlServerTaskPartition, SqlServerOffsetContext> taskOffsetContext = loader.load(
-                new SqlServerTaskPartition.Provider(connectorConfig, config).getPartitions());
+                new SqlServerTaskPartition.Provider(connectorConfig, config, dataConnection).getPartitions());
```

Comment:
It might be safer to use the `metadataConnection` here (see https://github.com/debezium/debezium/pull/917).